File: 0409-longest-palindrome/0409-longest-palindrome.py
class Solution(object):
    def longestPalindrome(self, s):
        dict = defaultdict(int)
        for c in s:
            dict[c]+=1;
            
        odd,ans=0,0
        for key in dict:
            if dict[key]%2==0:
                ans+=dict[key]
            else:
                if odd==0:
                    odd=1
                    ans+=dict[key]
                else:
                    ans+=dict[key]-1
        
        return ans

File: 0409-longest-palindrome/0409-longest-palindrome.java
class Solution {
    public int longestPalindrome(String s) {
        Map<Character,Integer> mp =new HashMap<>();
        
        for(char c: s.toCharArray()) {
            mp.put(c, mp.getOrDefault(c,0)+1);
        }
        int odd=0,ans=0;
        for(int value: mp.values()) {
            if(value%2!=0) {
                if(odd==0) {
                    ans+=value;
                    odd=1;
                }
                else ans+=value-1;
            }else ans+=value;
        }
        return ans;
    }
}